### PR TITLE
[feature-wip](multi-catalog) support iceberg union catalog, and add h…

### PR DESCRIPTION
**Pull request by wsjz**
support iceberg unified catalog framework, and add hms and rest catalog for the framework

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [x] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [x] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.41 seconds
 load time: 498 seconds
 storage size: 17122493381 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230118210813_clickbench_pr_83717.html

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/IcebergScanProvider.java [hunk 1]
```diff
@@ -52,23 +53,23 @@
...
 
 /**
  * A file scan provider for iceberg.
  */
-public class IcebergScanProvider extends HiveScanProvider {
+public class IcebergScanProvider extends QueryScanProvider {
 
     private static final int MIN_DELETE_FILE_SUPPORT_VERSION = 2;
     private final Analyzer analyzer;
+    private final IcebergSourceProvider delegate;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/IcebergScanProvider.java:69]:
```suggestion
    private final IcebergSourceProvider delegateProvider;
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/IcebergRestExternalCatalog.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        String restUri = catalogProperty.getProperties().getOrDefault(CatalogProperties.URI, "");
+        restProperties.put(CatalogProperties.URI, restUri);
+        RESTCatalog restCatalog = new RESTCatalog();
+        restCatalog.initialize(icebergCatalogType, restProperties);
+        catalog = restCatalog;
+    }
+
+    @Override
+    public List<String> listDatabaseNames() {
+        return ((RESTCatalog) catalog).listNamespaces().stream()
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/IcebergRestExternalCatalog.java:50]: hive catalog and rest catalog have different super class

File: fe/fe-core/src/main/java/org/apache/doris/datasource/IcebergHMSExternalCatalog.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        for (Map.Entry<String, String> entry : catalogProperties.entrySet()) {
+            conf.set(entry.getKey(), entry.getValue());
+        }
+        return conf;
+    }
+
+    @Override
+    public List<String> listDatabaseNames() {
+        return ((HiveCatalog) catalog).listNamespaces().stream()
+            .map(Namespace::toString).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/IcebergHMSExternalCatalog.java]: I think now we can just not supported dot separator in hive/spark/jdbc meta
For file or other customer catalog provide a hook to support the conversion of dot  to underline.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**